modules/sensor/src/rgb-depth/realsense/vpRealSense2.cpp

```diff
@@ -217,6 +217,197 @@ void vpRealSense2::acquire(unsigned char *const data_image, unsigned char *const
+             vpColVector *vel, vpColVector *acc, unsigned int *tracker_confidence, double *ts)
+{
+  auto data = m_pipe.wait_for_frames();
+
+  if(left != NULL)
+  {
+    auto left_fisheye_frame  = data.get_fisheye_frame(1);
+    getNativeFrameData(left_fisheye_frame, (*left).bitmap);
+  }
+
+  if(right != NULL)
+  {
+    auto right_fisheye_frame = data.get_fisheye_frame(2);
+    getNativeFrameData(right_fisheye_frame, (*right).bitmap);
+  }
+
+  auto pose_frame = data.first_or_default(RS2_STREAM_POSE);
+  auto pose_data  = pose_frame.as<rs2::pose_frame>().get_pose_data();
+
+  if(ts != NULL)
+    *ts = data.get_timestamp();
+
+  if(pose != NULL)
+  {
+    m_pos[0] = static_cast<double>(pose_data.translation.x);
+    m_pos[1] = static_cast<double>(pose_data.translation.y);
+    m_pos[2] = static_cast<double>(pose_data.translation.z);
+
+    m_quat[0] = static_cast<double>(pose_data.rotation.x);
+    m_quat[1] = static_cast<double>(pose_data.rotation.y);
+    m_quat[2] = static_cast<double>(pose_data.rotation.z);
+    m_quat[3] = static_cast<double>(pose_data.rotation.w);
+
+    *pose = vpHomogeneousMatrix(m_pos, m_quat);
+  }
+
+  if(vel != NULL)
+  {
+    (*vel)[0] = static_cast<double>(pose_data.velocity.x);
+    (*vel)[1] = static_cast<double>(pose_data.velocity.y);
+    (*vel)[2] = static_cast<double>(pose_data.velocity.z);
+    (*vel)[3] = static_cast<double>(pose_data.angular_velocity.x);
+    (*vel)[4] = static_cast<double>(pose_data.angular_velocity.y);
+    (*vel)[5] = static_cast<double>(pose_data.angular_velocity.z);
+  }
+
+    if(acc != NULL)
+  {
+    (*acc)[0] = static_cast<double>(pose_data.acceleration.x);
+    (*acc)[1] = static_cast<double>(pose_data.acceleration.y);
+    (*acc)[2] = static_cast<double>(pose_data.acceleration.z);
+    (*acc)[3] = static_cast<double>(pose_data.angular_acceleration.x);
+    (*acc)[4] = static_cast<double>(pose_data.angular_acceleration.y);
+    (*acc)[5] = static_cast<double>(pose_data.angular_acceleration.z);
+  }
+
+  *tracker_confidence = pose_data.tracker_confidence;
+}
+
+/*!
+  Acquire timestamped greyscale images, odometry and raw motion data from T265 Realsense device
+  at 30Hz.
+  \param left               : Left image.
+  \param right              : Right image.
+  \param pose               : Pointer to pose.
+  \param vel                : Pointer to velocity vector.
+  \param acc                : Pointer to acceleration vector.
+  \param raw_accel          : Pointer to Accelerometer raw data vector.
+  \param raw_gyro           : Pointer to Gyro raw data vector.
+  \param tracker_confidence : Pose estimation confidence (1: Low, 2: Medium, 3: High)
+  \param ts                 : Timestamp.
+*/
+void vpRealSense2::acquire(vpImage<unsigned char> *left, vpImage<unsigned char> *right, vpHomogeneousMatrix *pose,
+             vpColVector *vel, vpColVector *acc, vpColVector *raw_accel, vpColVector *raw_gyro,
+             unsigned int *tracker_confidence, double *ts)
+{
```

Comment:
Same remark concerning `tracker_confidence`
Does it make sense to give the user access to `raw_accel` and `raw_gyro` ?
Are `raw_accel` and `raw_gyro` data filtered to produce the content of `acc` ?
If useful we can keep, otherwise remove

Reply:
Yes. RS2_STREAM_ACCEL and RS2_STREAM_GYRO contains linear accelerations and angular velocities respectively. I thought we should have them as extra information.

Reply:
Ok, but it could be nice to investigate in order to see if the values of `raw_accel` and `raw_gyro` are exactly the same than the one in `acc`. If this is the case, please complete method description. The prefix `raw_` is confusing to me.

Reply:
Okay. I did some research and found that the stream `RS2_STREAM_ACCEL` provides linear acceleration from IMU at 62.5Hz and `RS2_STREAM_GYRO` provides angular velocities from IMU at 200Hz. But since we are wainting for frames `using wait_for_frames() `, we have the data at 30Hz (images frequency). 

But `acc` and `raw_acc` (now called `imu_acc`) should be different because `acc` is the result of visual inertial odometry and not directly from IMU. I will compare them and decide whether we should keep the new `acquire()` or not.

Reply:
`acc` and `imu_acc` contain different data.

Reply:
We should then rename `raw_gyro` into `gyro_vel`
and create dedicated functions to get these data at the max frequency
```
// Provides linear acceleration from IMU at 62.5Hz.
void getIMUAcceleration(vpColVector &imu_accel) const;
// Provides angular velocities from IMU at 200Hz.
void getGyroVelocities(vpColVector &gyro_vel) const;
```

Reply:
Okay. I will add these 2 functions.